src/approaches/oracle_approach.py

```diff
@@ -1001,3 +1010,183 @@ def _get_repeated_nextto_gt_nsrts() -> Set[NSRT]:
+            target_obj_type = type_name_to_type[target_obj_type_name]
+            target_obj = Variable("?targ", target_obj_type)
+
+            # Pick from ontop something
+            for surf_obj_type in sorted(env.types):
+                surf_obj = Variable("?surf", surf_obj_type)
+                parameters = [target_obj, agent_obj, surf_obj]
+                option_vars = [target_obj]
+                handempty = _get_lifted_atom("handempty", [])
+                targ_next_to = _get_lifted_atom(
+                    "reachable", [target_obj, agent_obj]
+                )
+                # targ_graspable = _get_lifted_atom(
+                #    "graspable", [target_obj]
+                # )
+                targ_holding = _get_lifted_atom("holding", [target_obj])
+                # preconditions = {handempty, targ_next_to, targ_graspable}
+                preconditions = {handempty, targ_next_to}
+                add_effects = {targ_holding}
+                delete_effects = {handempty}
+                nsrt = NSRT(
+                    f"{option.name}-{next(op_name_count)}",
+                    parameters,
+                    preconditions,
+                    add_effects,
+                    delete_effects,
+                    set(),
+                    option,
+                    option_vars,
+                    lambda s, r, o: grasp_obj_param_sampler(r),
+                )
+                nsrts.add(nsrt)
+
+        elif base_option_name == "PlaceOnTop":
+            assert len(option_arg_type_names) == 1
+            surf_obj_type_name = option_arg_type_names[0]
+            surf_obj_type = type_name_to_type[surf_obj_type_name]
+            surf_obj = Variable("?surf", surf_obj_type)
+
+            # We need to place the object we're holding!
+            for held_obj_types in sorted(env.types):
+                held_obj = Variable("?held", held_obj_types)
+                parameters = [held_obj, agent_obj, surf_obj]
+                option_vars = [surf_obj]
+                handempty = _get_lifted_atom("handempty", [])
+                held_holding = _get_lifted_atom("holding", [held_obj])
+                surf_next_to = _get_lifted_atom(
+                    "reachable", [surf_obj, agent_obj]
+                )
+                ontop = _get_lifted_atom("ontop", [held_obj, surf_obj])
+                preconditions = {held_holding, surf_next_to}
+                add_effects = {ontop, handempty}
+                delete_effects = {held_holding}
+                nsrt = NSRT(
+                  f"{option.name}-{next(op_name_count)}",
+                  parameters,
+                  preconditions,
+                  add_effects,
+                  delete_effects,
+                  set(),
+                  option,
+                  option_vars,
+                  lambda s, r, o: place_ontop_obj_pos_sampler(  # type: ignore
+                    env.behavior_env,  # type: ignore
```

Comment:
instead of passing in `env.behavior_env` here, if you need access to an arbitrary behavior env instance within the place sampler, then can you use `env = get_env_instance("behavior")` within the sampler itself?

Reply:
I don't need access to an arbitrary behavior env instance, but rather the specific behavior env currently tied to `env`

Reply:
but `env` is just an arbitrary instance in the context of this function. what do you need exactly? if you need it to match up with `s`, then maybe you should use `s.simulator_state`?

Reply:
ah - good point about `env` being arbitrary. I ended up realizing we don't actually need the env, so this is now fixed!